pkg/scale/decode.go

```diff
@@ -108,6 +109,20 @@ type decodeState struct {
 }
 
 func (ds *decodeState) unmarshal(dstv reflect.Value) (err error) {
+	// Handle BTreeMap type separately for the following reasons:
+	// 1. BTreeMap is a generic type, so we can't use the normal type switch
+	// 2. We cannot use BTreeCodec because we are comparing the type of the dstv.Interface() in the type switch
+	if isBTree(dstv.Type()) {
```

Comment:
Just for my own clarification, once we determine that it is either `btree.Btree` or `btree.Map` we decode in the same fashion?

Reply:
Yes, since both are implementing the same interface, it just calls `Decode()`